internal/storage/bucket_handle_test.go

```diff
@@ -256,3 +256,51 @@ func (t *BucketHandleTest) TestCreateObjectMethodWhenGivenGenerationObjectNotExi
 	AssertEq(nil, obj)
 	AssertTrue(strings.Contains(err.Error(), "Error 412: Precondition failed"))
 }
+
+func (t *BucketHandleTest) TestListObjectMethodWithValidPrefix() {
+	obj, err := t.bucketHandle.ListObjects(context.Background(),
+		&gcs.ListObjectsRequest{
+			Prefix:                   "gcsfuse/",
+			Delimiter:                "/",
+			IncludeTrailingDelimiter: true,
+			ContinuationToken:        "ContinuationToken",
+			MaxResults:               7,
+			ProjectionVal:            0,
+		})
+
+	AssertNe(nil, obj.Objects)
+	AssertNe(nil, obj.CollapsedRuns)
+	AssertEq(nil, err)
```

Comment:
you should always check for error first. 
If error is not present, then we check for data. No point in checking for error after data check